README.md

```diff
@@ -13,68 +13,125 @@ npm install
+#### Allow redirects
 ```
-KnowSheet> JSONParse(HTTP(GET("http://httpbin.org/get?query=1")).body).args
+KnowSheet> HTTP(GET("http://httpbin.org/redirect/5").AllowRedirects()).body
 ```
 
-POST a JSON-encoded object:
+#### Provide a custom `User-Agent`
 ```
-KnowSheet> JSONParse(HTTP(POST("http://httpbin.org/post", DemoObject())).body).json
+KnowSheet> HTTP(GET("http://httpbin.org/get").UserAgent("KnowSheet-REPL/1.0.0")).body
 ```
-<sup>The syntax mimics C++ Bricks exactly, as long as `DemoObject` is defined as a serializable type.</sup>
 
-Allow redirects:
+#### Provide extra HTTP headers
+For `GET` requests:
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/redirect/5").AllowRedirects()).body
+KnowSheet> HTTP(GET("http://httpbin.org/get?query=1", HTTPHeaders().Set("Custom", "Header"))).body
 ```
 
-Provide a custom User-Agent:
+For `POST` requests:
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/get").UserAgent("KnowSheet-REPL/1.0.0")).body
+KnowSheet> HTTP(POST("http://httpbin.org/post", "BODY", "text/plain", HTTPHeaders().Set("Custom", "Header"))).body
 ```
 
-Provide extra HTTP headers:
+### HTTP server
+
+* Bricks maintains a single server per port.
+* Each server accepts connections since start till the process exits.
+* There can only be a single handler for a given path.
+* The calls to the server methods can be chained.
+* The handler lambda syntax in the REPL mimics C++11 to a certain extent -- no full lambda support guaranteed.
+
+#### Start an HTTP server
 ```
-KnowSheet> HTTP(GET("http://httpbin.org/get?query=1", HTTPHeaders().Set("Custom", "Header"))).body
-KnowSheet> HTTP(POST("http://httpbin.org/post", "BODY", "text/plain", HTTPHeaders().Set("Custom", "Header"))).body
+KnowSheet> HTTP(2015)
+// KnowSheet Bricks HTTPServer at port 2015
+```
+
+#### Register an endpoint on an HTTP server
+```
+KnowSheet> HTTP(2015).Register("/ping", [](Request r) { r("pong"); })
+// KnowSheet Bricks HTTPServer at port 2015
+KnowSheet> HTTP(GET("http://localhost:2015/ping")).body
+--------------------------------------------------------------------------------
+pong
+--------------------------------------------------------------------------------
+(32ms)
 ```
 
-Chain requests:
+#### Unregister an endpoint from an HTTP server
+```
+KnowSheet> HTTP(2015).UnRegister("/ping")
+```
+
+#### Use the request data in an HTTP server handler
+```
+KnowSheet> HTTP(2015).Register("/test", [](Request r){ r(r.timestamp + " " + r.method + " " + r.url.ComposeURL() + "\n" + r.body); })
+// KnowSheet Bricks HTTPServer at port 2015
+KnowSheet> HTTP(POST("localhost:2015/test", "BODY")).body
+--------------------------------------------------------------------------------
+1426024237701 POST /test
+BODY
+--------------------------------------------------------------------------------
+(12ms)
+```
+
+### JSON
+
+#### POST a JSON-encoded instance of a serializable type
+```
+KnowSheet> HTTP(POST("http://httpbin.org/post", DemoObject())).body
```

Comment:
Right, but here we don't really define a new type, just providing it along with the parameters, aren't we?